File: BinarySearch.cs
using System;

public class BinarySearchClass {
// O(logn) time and O(1) space
  public static int BinarySearch(int[] array, int target) {
    int first =0;
    int last = array.Length-1;
    int mid=0;
    while(first<=last)
    {
    mid = Math.Abs(first + last)/2;
    if(target<array[mid])
        last = mid-1;
    else if(target> array[mid])
        first = mid+1;
    else
        return mid;
    }
    return -1;
  }
}

File: BinarySearch.py
class BinarysearchPython:
    # O(log n) time and O(1) space
    def search(self, nums: List[int], target: int) -> int:
        min, max = 0, len(nums) -1
        while min<=max:
            mid = int(min+(max-min)/2)
            if target==nums[mid]:
                return mid
            elif target<nums[mid]:
                max=mid-1
            else:
                min = mid+1
        return -1
